jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/OdeTimDataCreatorHelper.java

```diff
@@ -0,0 +1,85 @@
+package us.dot.its.jpo.ode.coder;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import us.dot.its.jpo.ode.context.AppContext;
+import us.dot.its.jpo.ode.model.OdeMsgMetadata;
+import us.dot.its.jpo.ode.model.OdeTimData;
+import us.dot.its.jpo.ode.model.OdeTimMetadata;
+import us.dot.its.jpo.ode.model.OdeTimPayload;
+import us.dot.its.jpo.ode.model.ReceivedMessageDetails;
+import us.dot.its.jpo.ode.model.RxSource;
+import us.dot.its.jpo.ode.plugin.j2735.travelerinformation.TravelerInformation;
+import us.dot.its.jpo.ode.util.JsonUtils;
+import us.dot.its.jpo.ode.util.XmlUtils;
+import us.dot.its.jpo.ode.util.XmlUtils.XmlUtilsException;
+
+/**
+ * Helper class for deserializing TIM messages in XML/XER format into POJOs.
+ */
+public class OdeTimDataCreatorHelper {
+
+  /**
+   * Deserializes XML/XER from the UDP decoded pipeline.
+   *
+   * @param consumedData The XML/XER as a String.
+   */
+  public static OdeTimData createOdeTimDataFromDecoded(String consumedData) 
+      throws XmlUtilsException {
+    ObjectNode consumed = XmlUtils.toObjectNode(consumedData);
+
+    JsonNode metadataNode = consumed.findValue(AppContext.METADATA_STRING);
+    if (metadataNode instanceof ObjectNode) {
+      ObjectNode object = (ObjectNode) metadataNode;
+      object.remove(AppContext.ENCODINGS_STRING);
+
+      // Map header file does not have a location and use predefined set required
+      // RxSource
+      ReceivedMessageDetails receivedMessageDetails = new ReceivedMessageDetails();
+      receivedMessageDetails.setRxSource(RxSource.NA);
+      ObjectMapper objectMapper = new ObjectMapper();
+      JsonNode jsonNode;
+      try {
+        jsonNode = objectMapper.readTree(receivedMessageDetails.toJson());
+        object.set(AppContext.RECEIVEDMSGDETAILS_STRING, jsonNode);
+      } catch (JsonProcessingException e) {
+        e.printStackTrace();
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
```

Comment:
suggestion: ensure we are capturing the errors in our logs not via standard out (what e.printStackTrace() does). Note that the catch blocks were collapsed. We don't do anything different in the two exception catch blocks, so it's safe to unify them under a single generic Exception catch block
```suggestion
      } catch (Exception e) {
        log.error("Failed to read JSON node: {}", e.getMessage());
      }
```

Reply:
I added this one as well. I also made the exception generic like this to simplify the code.